#ifndef WORKOUTSVIEW_H
#define WORKOUTSVIEW_H

#include <QWidget>
#include <QMap> // <--- WA呕NE: Dodane do obsugi link贸w
#include "./AI/aiworkoutgenerator.h"
#include "./AI/aiworkoutplan.h"
#include "./Model/trainingmanager.h"
#include "./Model/usermodel.h"
#include "./Model/workout.h"

namespace Ui {
class WorkoutsView;
}

class WorkoutsView : public QWidget
{
    Q_OBJECT

public:
    explicit WorkoutsView(QWidget *parent = nullptr);
    ~WorkoutsView();

    void setUserData(UserModel *currentUser);

private slots:
    void onGenerateAiWorkoutClicked();
    void onAiPlanReady(const AIWorkoutPlan &plan);
    void onAddExerciseClicked();
    void onSaveWorkoutClicked();

    // Nowy slot do ukrywania p贸l (Cardio vs Strength)
    void toggleInputFields();

    // Slot usuwania z tabeli (wywoywany przez przycisk w tabeli)
    void removeExerciseAt(int index);

    //  NOWY SLOT DO WIDEO
    void onShowVideoClicked();

private:
    void loadExercises(const QString &type);

    // Nowa funkcja do odwie偶ania tabeli
    void updateExerciseTable();

    //  Funkcja pomocnicza (偶eby nie rusza TrainingManagera)
    QString getExerciseNameById(int id);

    //  Inicjalizacja link贸w
    void initVideoLinks();

    TrainingManager *trainingManager;
    QList<WorkoutExercisePlan> currentPlan;
    Ui::WorkoutsView *ui;

    UserModel *currentUser = nullptr;

    //  Mapa link贸w
    QMap<QString, QString> exerciseLinks;
};

#endif // WORKOUTSVIEW_H
